service/controller/app/v1/resource/chartoperator/resource.go

```diff
@@ -0,0 +1,196 @@
+package chartoperator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/helmclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/helm/pkg/helm"
+
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	appcatalogkey "github.com/giantswarm/app-operator/service/controller/appcatalog/v1/key"
```

Comment:
I think when controllers import different key packages something went really wrong in the design. Let's please not do that. I saw this in `cluster-operator` everywhere and it is just painful to work with because it is so complicated.

Reply:
Agree I think everything should be in `service/controller/app/v1/key`.  As the appcatalog controller has no resources and is just for creating the CRD.